tests/test_worker.py

```diff
@@ -449,3 +450,55 @@ class Foo:
     assert await j1.status() == JobStatus.complete
 
     assert await arq_redis.enqueue_job('foobar', _job_id='job_id') is None
+
+
+async def test_queue_read_limit_equals_max_jobs(arq_redis: ArqRedis, worker):
+    for _ in range(4):
+        await arq_redis.enqueue_job('foobar')
+
+    assert await arq_redis.zcard(default_queue_name) == 4
+    worker: Worker = worker(functions=[foobar], max_jobs=2)
+    worker.pool = await create_pool(worker.redis_settings)
```

Comment:
why do you need this?

Reply:
@samuelcolvin My understanding was that the pool is initialized in `Worker.main`, which we don't call in this test. It seems that it's initialized separately too, so I removed this line.

Reply:
I think the point is that the `worker` fixture uses the standard redis connection, thus this isn't required.